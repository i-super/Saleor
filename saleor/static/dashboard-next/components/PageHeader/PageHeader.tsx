import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import * as React from "react";

import ExtendedPageHeader from "../ExtendedPageHeader";
import Skeleton from "../Skeleton";

const decorate = withStyles(
  theme => ({
    title: {
      flex: 1,
      paddingBottom: theme.spacing.unit * 2
    }
  }),
  { name: "PageHeader" }
);

interface PageHeaderProps {
  className?: string;
  title?: string;
  onBack?();
}

const PageHeader = decorate<PageHeaderProps>(
  ({ children, classes, className, onBack, title }) => (
    <ExtendedPageHeader
      className={className}
      onBack={onBack}
      title={
        <Typography className={classes.title} variant="title">
          {title !== undefined ? title : <Skeleton style={{ width: "10em" }} />}
        </Typography>
      }
    >
      {children}
    </ExtendedPageHeader>
  )
);
PageHeader.displayName = "PageHeader";
export default PageHeader;
