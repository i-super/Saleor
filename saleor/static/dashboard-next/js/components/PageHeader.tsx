import ArrowBackIcon from "material-ui-icons/ArrowBack";
import CloseIcon from "material-ui-icons/Close";
import Divider from "material-ui/Divider";
import IconButton from "material-ui/IconButton";
import { withStyles } from "material-ui/styles";
import Typography from "material-ui/Typography";
import * as React from "react";
import { Link } from "react-router-dom";
import Skeleton from "./Skeleton";

const decorate = withStyles(theme => ({
  action: {
    flex: "0 0 auto",
    marginRight: theme.spacing.unit * -2
  },
  grid: {
    padding: theme.spacing.unit * 2
  },
  menuButton: {
    flex: "0 0 auto",
    marginLeft: theme.spacing.unit * -2,
    marginRight: theme.spacing.unit * 3
  },
  root: theme.mixins.gutters({
    alignItems: "center",
    display: "flex"
  }),
  subtitle: {
    alignItems: "center" as "center",
    display: "flex",
    marginBottom: theme.spacing.unit * 2
  },
  title: {
    flex: 1,
    paddingBottom: theme.spacing.unit * 2,
    paddingTop: theme.spacing.unit * 2
  }
}));

interface PageHeaderProps {
  title: string;
  onBack?();
  onCancel?();
}

const PageHeader = decorate<PageHeaderProps>(
  ({ children, classes, onBack, onCancel, title }) => (
    <div className={classes.root}>
      {onBack && (
        <IconButton
          color="inherit"
          className={classes.menuButton}
          onClick={onBack}
        >
          <ArrowBackIcon />
        </IconButton>
      )}
      {onCancel && (
        <IconButton
          color="inherit"
          className={classes.menuButton}
          onClick={onCancel}
        >
          <CloseIcon />
        </IconButton>
      )}
      <Typography className={classes.title} variant="title">
        {title !== undefined ? title : <Skeleton style={{ width: "10em" }} />}
      </Typography>
      <div className={classes.action}>{children}</div>
    </div>
  )
);

export default PageHeader;
