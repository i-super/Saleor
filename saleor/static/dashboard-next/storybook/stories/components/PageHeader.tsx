import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";
import { storiesOf } from "@storybook/react";
import * as React from "react";

import PageHeader from "../../../components/PageHeader";

storiesOf("Generics / PageHeader", module)
  .add("without title", () => <PageHeader />)
  .add("with title", () => <PageHeader title="Lorem ipsum" />)
  .add("with title icon bar", () => (
    <PageHeader title="Lorem ipsum">
      <IconButton>
        <DeleteIcon />
      </IconButton>
    </PageHeader>
  ));
