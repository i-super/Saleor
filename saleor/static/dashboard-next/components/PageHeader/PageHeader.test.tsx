import DeleteIcon from "material-ui-icons/Delete";
import IconButton from "material-ui/IconButton";
import * as React from "react";
import * as renderer from "react-test-renderer";

import PageHeader from "./";

describe("<PageHeader />", () => {
  it("renders without title", () => {
    const component = renderer.create(<PageHeader />);
    expect(component).toMatchSnapshot();
  });
  it("renders with title", () => {
    const component = renderer.create(<PageHeader title="Lorem ipsum" />);
    expect(component).toMatchSnapshot();
  });
  it("renders with title and back button", () => {
    const component = renderer.create(
      <PageHeader title="Lorem ipsum" onBack={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with title icon bar", () => {
    const component = renderer.create(
      <PageHeader title="Lorem ipsum">
        <IconButton>
          <DeleteIcon />
        </IconButton>
      </PageHeader>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders with title, back button and icon bar", () => {
    const component = renderer.create(
      <PageHeader title="Lorem ipsum" onBack={jest.fn()}>
        <IconButton>
          <DeleteIcon />
        </IconButton>
      </PageHeader>
    );
    expect(component).toMatchSnapshot();
  });
});
