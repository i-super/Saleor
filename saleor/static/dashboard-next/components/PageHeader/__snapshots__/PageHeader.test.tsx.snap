// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> renders with title 1`] = `
<div
  className="Component-root-4"
>
  <h2
    className="MuiTypography-root-7 MuiTypography-title-13 Component-title-6"
  >
    Lorem ipsum
  </h2>
  <div
    className="Component-action-1"
  />
</div>
`;

exports[`<PageHeader /> renders with title and back button 1`] = `
<div
  className="Component-root-4"
>
  <button
    className="MuiButtonBase-root-38 MuiIconButton-root-32 MuiIconButton-colorInherit-33 Component-menuButton-3"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-37"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-41"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </g>
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-47"
    />
  </button>
  <h2
    className="MuiTypography-root-7 MuiTypography-title-13 Component-title-6"
  >
    Lorem ipsum
  </h2>
  <div
    className="Component-action-1"
  />
</div>
`;

exports[`<PageHeader /> renders with title icon bar 1`] = `
<div
  className="Component-root-4"
>
  <h2
    className="MuiTypography-root-7 MuiTypography-title-13 Component-title-6"
  >
    Lorem ipsum
  </h2>
  <div
    className="Component-action-1"
  >
    <button
      className="MuiButtonBase-root-38 MuiIconButton-root-32"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-37"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-41"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </g>
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-47"
      />
    </button>
  </div>
</div>
`;

exports[`<PageHeader /> renders with title, back button and icon bar 1`] = `
<div
  className="Component-root-4"
>
  <button
    className="MuiButtonBase-root-38 MuiIconButton-root-32 MuiIconButton-colorInherit-33 Component-menuButton-3"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-37"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-41"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </g>
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-47"
    />
  </button>
  <h2
    className="MuiTypography-root-7 MuiTypography-title-13 Component-title-6"
  >
    Lorem ipsum
  </h2>
  <div
    className="Component-action-1"
  >
    <button
      className="MuiButtonBase-root-38 MuiIconButton-root-32"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiIconButton-label-37"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-41"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </g>
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-47"
      />
    </button>
  </div>
</div>
`;

exports[`<PageHeader /> renders without title 1`] = `
<div
  className="Component-root-4"
>
  <h2
    className="MuiTypography-root-7 MuiTypography-title-13 Component-title-6"
  >
    <span
      className="Component-skeleton-31"
      style={
        Object {
          "width": "10em",
        }
      }
    >
      â€Œ
    </span>
  </h2>
  <div
    className="Component-action-1"
  />
</div>
`;
